runner/runners/polling_test.go

```diff
@@ -149,3 +149,27 @@ func TestPolling_ErrorPolling(t *testing.T) {
 	ex.Resume()
 
 }
+
+func TestPollingWorker_PollFrequency(t *testing.T) {
+	defer teardown(t)
+	sim := execers.NewSimExecer()
+
+	outputCreator, err := NewHttpOutputCreator("")
+	if err != nil {
+		panic(err)
+	}
+
+	filerMap := runner.MakeRunTypeMap()
+	filerMap[runner.RunTypeScoot] = snapshot.FilerAndInitDoneCh{Filer: snapshots.MakeInvalidFiler(), IDC: nil}
+
+	r := NewSingleRunner(sim, filerMap, outputCreator, nil, stats.NopDirsMonitor, runner.EmptyID)
+	args := []string{"pause", "complete 0"}
+	startTime := time.Now()
+	expectedTime := startTime.Add(10 * time.Microsecond)
+	firstRun := run(t, r, args)
+	assertWait(t, r, firstRun, pending(), args...)
+
+	assert.WithinDuration(t, expectedTime, time.Now(), 1*time.Millisecond)
```

Comment:
I'm trying to figure out exactly what is being tested here - looks like it is verifying that one poll interval has elapsed?  Which may not verify that the poll interval is increasing (I could be missing something). This is also a very high resolution time delta, so I'm wondering if this will end up being flaky depending on where it's being run.

FWIW I don't see any great way to test this behavior. Bad practice generally but I'm not averse to skipping if we can't make one in this instance.

Reply:
jbruno/testPollingFreq branch shows how we could instrument a test to expose the polling frequency for test assertions (warning ... it is not pretty)